### PR TITLE
Companion for substrate 14647

### PR DESCRIPTION
**Pull request by davxy**
Companion for https://github.com/paritytech/substrate/pull/14647

### CODE REVIEW

File: client/consensus/aura/src/collators/lookahead.rs [hunk 1]
```diff
@@ -106,9 +106,9 @@ pub async fn run<Block, P, BI, CIDP, Client, Backend, RClient, SO, Proposer, CS>
 	Proposer: ProposerInterface<Block, Transaction = BI::Transaction>,
 	Proposer::Transaction: Sync,
 	CS: CollatorServiceInterface<Block>,
-	P: Pair + Send + Sync,
-	P::Public: AppPublic + Hash + Member + Encode + Decode,
-	P::Signature: TryFrom<Vec<u8>> + Hash + Member + Encode + Decode,
+	P: Pair,
+	P::Public: AppPublic + Member + Codec,
```
Comment (skunert) [inline client/consensus/aura/src/collators/lookahead.rs:110]:
```suggestion
	P::Public: AppPublic + Member,
```

Comment (skunert) [inline client/consensus/aura/src/collators/lookahead.rs:110]:
```suggestion
	P::Public: AppPublic + Member,
```

File: client/consensus/aura/src/collators/basic.rs [hunk 2]
```diff
@@ -87,9 +87,9 @@ pub async fn run<Block, P, BI, CIDP, Client, RClient, SO, Proposer, CS>(
 	Proposer: ProposerInterface<Block, Transaction = BI::Transaction>,
 	Proposer::Transaction: Sync,
 	CS: CollatorServiceInterface<Block>,
-	P: Pair + Send + Sync,
-	P::Public: AppPublic + Hash + Member + Encode + Decode,
-	P::Signature: TryFrom<Vec<u8>> + Hash + Member + Encode + Decode,
+	P: Pair,
+	P::Public: AppPublic + Member + Codec,
```
Comment (skunert) [inline client/consensus/aura/src/collators/basic.rs:91]:
```suggestion
	P::Public: AppPublic + Member,
```

File: client/consensus/aura/src/collator.rs [hunk 3]
```diff
@@ -98,9 +98,9 @@ where
 	Proposer: ProposerInterface<Block, Transaction = BI::Transaction>,
 	Proposer::Transaction: Sync,
 	CS: CollatorServiceInterface<Block>,
-	P: Pair + Send + Sync,
-	P::Public: AppPublic + Hash + Member + Encode + Decode,
-	P::Signature: TryFrom<Vec<u8>> + Hash + Member + Encode + Decode,
+	P: Pair,
+	P::Public: AppPublic + Member + Codec,
```
Comment (skunert) [inline client/consensus/aura/src/collator.rs:102]:
```suggestion
	P::Public: AppPublic + Member,
```

**Review by skunert — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3271985

**Pull request merged**